app/models/hackney/cautionary_contact.rb

```diff
@@ -0,0 +1,26 @@
+class Hackney::CautionaryContact
+  include ActiveModel::Model
+
+  attr_accessor :property_reference, :contact_number, :title, :forename, :surname, :caller_notes, :alert_code
+
+  def self.find(property_reference)
+    response = HackneyAPI::RepairsClient.new.get_cautionary_contact(property_reference)
+    if response['results'].any?
+      build(response['results'].first)
+    else
+      nil
+    end
+  end
+
+  def self.build(attributes)
```

Comment:
I know you are following the pattern that is already present on the code, but let's not do that.

ActiveRecord has the same method, like `model.associations.build(attributes)`, but this one clearly doesn't follow the same interface.

We should also look at the `RepairRequest` class and discuss how this can be refactored.

Reply:
What about naming this something other than `build`, like `new_from_api`. The `build` name is part of ActiveRecord's interface and it's confusing to have it behaving differently.

Reply:
👍